app/(you)/preferences.tsx

```diff
@@ -0,0 +1,108 @@
+import * as SecureStore from 'expo-secure-store';
+import i18n from 'i18next';
+import { useColorScheme } from 'nativewind';
+import React, { useContext, useEffect, useRef, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { View } from 'react-native';
+
+import { SelectList } from '@/components';
+import SafeView from '@/components/ui/box/SafeView';
+import CustomCheckbox from '@/components/ui/checkbox/CustomCheckbox';
+import CustomHeader from '@/components/ui/header/CustomHeader';
+import CustomSwitch from '@/components/ui/switch/CustomSwitch';
+import CustomTable from '@/components/ui/table/CustomTable';
+import { LANGUAGE_KEY, MODE_KEY } from '@/constants/Storage';
+import { GlobalContext } from '@/context/GlobalContext';
+
+export default function Preferences() {
+  const { t } = useTranslation();
+  const { colorScheme, setColorScheme } = useColorScheme();
+  const { preferences, setPreferences } = useContext(GlobalContext);
+  const [isSwitchOn, setSwitchOn] = useState(colorScheme === 'dark');
+  const [isCheckboxChecked, setCheckboxChecked] = useState(preferences.mode === 'system');
+
+  const hasPageBeenRendered = useRef({
+    checkbox: false,
+    switch: false,
+    colorScheme: false,
+    preferences: false,
+  });
+
+  useEffect(() => {
+    if (hasPageBeenRendered.current['checkbox']) {
+      if (isCheckboxChecked) {
+        setPreferences({ ...preferences, mode: 'system' });
+        setColorScheme('system');
+      } else {
+        setPreferences({ ...preferences, mode: colorScheme });
+      }
+    }
+    hasPageBeenRendered.current['checkbox'] = true;
+  }, [isCheckboxChecked]);
+
+  useEffect(() => {
+    if (hasPageBeenRendered.current['colorScheme']) {
+      if (isCheckboxChecked) {
+        setSwitchOn(colorScheme === 'dark');
+      }
+    }
+
+    hasPageBeenRendered.current['colorScheme'] = true;
+  }, [colorScheme]);
+
+  useEffect(() => {
+    if (hasPageBeenRendered.current['switch']) {
+      if (!isCheckboxChecked) {
+        const scheme = isSwitchOn ? 'dark' : 'light';
+        setColorScheme(scheme);
+        setPreferences({ ...preferences, mode: scheme });
+      }
+    }
+    hasPageBeenRendered.current['switch'] = true;
+  }, [isSwitchOn]);
+
+  useEffect(() => {
+    if (hasPageBeenRendered.current['preferences']) {
+      SecureStore.setItem(MODE_KEY, preferences.mode);
+    }
+    hasPageBeenRendered.current['preferences'] = true;
+  }, [preferences.mode]);
+
+  async function setLanguage(rawLanguage: string) {
+    const language: 'pl' | 'en' = rawLanguage === 'pl' || rawLanguage === 'en' ? rawLanguage : 'en';
```

Comment:
i18n have automatic fallback to eng we could just set async language to rawlanguage and then return langauge from i18n.langauge to set in our preferences. Key point for not hangle only 'pl' | 'en'